### PR TITLE
many changes

### PR DESCRIPTION
**Pull request by solodyagin**
Significant changes:

  *  horizontal scrolling;
  * add bar when double click on row;
  * magnetic effect (snap to grid);
  * tooltip format;
  * additional styles for row and label;
  * etc.

Demo: https://tenrok.github.io/vue-ganttastic/

P.S. Sorry for my bad English

### CODE REVIEW

**Comment by InfectoOne** [comment]
Wow, I am impressed by this contribution! I will take a look at this. An upgrade to Vue 3 will be released soon, but we could merge this into the final release of the Vue 2 version.

File: lib/components/GGanttGrid.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    precision: { type: String },
+    timeCount: { type: Number },
+    gridSize: { type: Number }
+  },
+
+  computed: {
+    allChildPoints() {
+      let start = moment(this.chartStart)
+      let end = moment(this.chartEnd)
+      let res = []
```
Comment (diegood) [inline lib/components/GGanttGrid.vue:45]:
```suggestion
      const start = moment(this.chartStart)
      const end = moment(this.chartEnd)
      const res = []
```
At no time is the reference modified, therefore it does not have to be a LET.

File: lib/components/GGanttRow.vue [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    onDragleave() {
+      this.$refs['g-gantt-row'].style.backgroundColor = null
+    },
+
+    onDrop(e) {
+      let barContainer = this.$refs.barContainer.getBoundingClientRect()
+      let xPos = e.clientX - barContainer.left
+      let timeDiffFromStart = (xPos / barContainer.width) * this.timeCount
+      let time = moment(this.chartStart).add(timeDiffFromStart, this.timeUnit)
+      let bar = this.localBars.find(bar =>
```
Comment (diegood) [inline lib/components/GGanttRow.vue:166]:
```suggestion
      const barContainer = this.$refs.barContainer.getBoundingClientRect()
      const xPos = e.clientX - barContainer.left
      const timeDiffFromStart = (xPos / barContainer.width) * this.timeCount
      const time = moment(this.chartStart).add(timeDiffFromStart, this.timeUnit)
      const bar = this.localBars.find(bar =>
```

File: lib/components/GGanttTimeaxis.vue [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+  methods: {
+    initAxis() {
+      switch (this.precision) {
+        case 'day':
+          this.initAxisDaysAndHours()
+          break
+        case 'month':
+          this.initAxisMonthsAndDays()
+          break
+      }
```
Comment (diegood) [inline lib/components/GGanttTimeaxis.vue:120]:
```suggestion
     this.precision ===  'day'? this.initAxisDaysAndHours() : this.initAxisMonthsAndDays()
```

**Review by GweltazNiquel — APPROVED** [review]

**Comment by InfectoOne** [comment]
Can you provide a brief overview of all significant changes introduced here?

**Comment by solodyagin** [comment]
Significant changes:
- horizontal scrolling;
- add bar when double click on row;
- magnetic effect (snap to grid);
- tooltip format;
- additional styles for row and label;
- etc.

Demo: https://tenrok.github.io/vue-ganttastic/

P.S. Sorry for my bad English

**Pull request closed without merging**